lib/logstash/filters/mutate.rb

```diff
@@ -233,6 +245,18 @@ def remove(event)
   end # def remove
 
   private
+  def select(event)
+    replacement = LogStash::Event.new
+
+    @select.each do |field|
+      field = event.sprintf(field)
+      next unless event.include?(field)
+      replacement[field] = event[field]
+    end
+    event.overwrite replacement
```

Comment:
As I understand the behaviour of this, this will actually override the previous event private variables, for now. Keeping the former ```@timestamp``` and ```@version```, not sure if also keep everything that start with ```@``` would be nice too. Is something required, otherwise events where people might use this function are going to be misleading. More details on how event.override works: https://github.com/elastic/logstash/blob/master/lib/logstash/event.rb#L167-L177